### PR TITLE
ISSUE #19

### PR DESCRIPTION
**Pull request by Zacklane**
User system implemented, based on a symfony general security component.

### CODE REVIEW

File: app/Resources/views/base.html.twig [hunk 1]
```diff
@@ -22,6 +22,8 @@
                     <li><a href="{{ url('dashboard') }}">Dashboard</a></li>
                     <li><a href="{{ url('transactions') }}">Transactions</a></li>
                     <li><a href="{{ url('licenses') }}">Licenses</a></li>
+                    <li><a href="{{ path('logout') }}"><span class="aui-icon aui-icon-small aui-iconfont-locked"></span> Lock
```
Comment (thecatontheflat) [inline app/Resources/views/base.html.twig]: Please keep the formatting right:
```
<li>
   <a>...</a>
</li>
```

File: app/Resources/views/security/login.html.twig [hunk 2]
```diff
@@ -0,0 +1,79 @@
+{% import ":other:macro.html.twig" as macro %}
```
Comment (thecatontheflat) [inline app/Resources/views/security/login.html.twig:1]: Is it possible to extend the same twig layout as the rest of the views?

File: app/config/security.yml [hunk 3]
```diff
@@ -1,52 +1,34 @@
...
+                target: /
 
-    # with these settings you can restrict or allow access for different parts
-    # of your application based on roles, ip, host or methods
-    # http://symfony.com/doc/current/cookbook/security/access_control.html
     access_control:
-        #- { path: ^/login, roles: IS_AUTHENTICATED_ANONYMOUSLY, requires_channel: https }
\ No newline at end of file
+        - { path: ^/login, roles: IS_AUTHENTICATED_ANONYMOUSLY }
+        - { path: ^/secured, role: ROLE_ADMIN }
```
Comment (thecatontheflat) [inline app/config/security.yml]: Please change naming —  `secured` to `protected`

**Comment by thecatontheflat** [comment]
Fixes #19

File: app/Resources/views/security/login.html.twig [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                                <div class="field-group">
+                                    <label for="username">Username<span class="aui-icon icon-required">(required)</span></label>
+                                    <input type="text" id="username" name="_username" value="{{ last_username }}"
+                                           required="required"
+                                           class="text medium-field"/>
+                                </div>
+
+                                <div class="field-group">
+                                    <label for="password">Password<span class="aui-icon icon-required">(required)</span></label>
+                                    <input type="password" id="password" name="_password" required="required"
```
Comment (thecatontheflat) [inline app/Resources/views/security/login.html.twig]: Please keep one tag formatting in the same line

Comment (thecatontheflat) [reply app/Resources/views/security/login.html.twig]: Applies to similar places as well

File: app/Resources/views/security/login.html.twig [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+            </div>
+        </div>
+    </section>
+    <footer id="footer" role="contentinfo">
+    </footer>
+</div>
+<script src="{{ asset('assets/jquery/dist/jquery.min.js') }}"></script>
+<script src="{{ asset('assets/aui/js/aui.min.js') }}"></script>
+<script src="{{ asset('assets/vendor/highcharts.js') }}"></script>
+{#<script src="{{ asset('assets/aui/js/aui-experimental.min.js') }}"></script>#}
```
Comment (thecatontheflat) [inline app/Resources/views/security/login.html.twig]: Please remove commented block

File: composer.lock [hunk 6]
```diff
@@ -4,7 +4,6 @@
         "Read more about it at https://getcomposer.org/doc/01-basic-usage.md#composer-lock-the-lock-file",
```
Comment (thecatontheflat) [inline composer.lock]: Please remove this file

File: src/AppBundle/Controller/CompanyController.php [hunk 7]
```diff
@@ -11,7 +11,7 @@
 class CompanyController extends Controller
 {
     /**
-     * @Route("/companies", name="companies")
+     * @Route("/secured/companies", name="companies")
```
Comment (thecatontheflat) [inline src/AppBundle/Controller/CompanyController.php]: Change to `secured` to `protected`

File: src/AppBundle/Controller/DashboardController.php [hunk 8]
```diff
@@ -8,10 +8,23 @@
...
+        if($user) {
+            return $this->redirectToRoute('dashboard');
+        }
+        else {
+            return $this->redirectToRoute('login');
+        }
+    }
     /**
-     * @Route("/", name="dashboard")
+     * @Route("/secured/dashboard", name="dashboard")
```
Comment (thecatontheflat) [inline src/AppBundle/Controller/DashboardController.php]: Same here `secured` => `protected`

File: src/AppBundle/Controller/SecurityController.php [hunk 9]
```diff
@@ -0,0 +1,32 @@
+<?php
+// src/AppBundle/Controller/SecurityController.php
```
Comment (thecatontheflat) [inline src/AppBundle/Controller/SecurityController.php]: Remove commented lines

**Review by thecatontheflat — CHANGES_REQUESTED** [review]

File: src/AppBundle/Controller/DashboardController.php (hunk 8 above)
Comment (Zacklane) [reply src/AppBundle/Controller/DashboardController.php]: Secured changed to protected, as you've requested, in all the routes and firewall rule.

File: app/Resources/views/security/login.html.twig (hunk 5 above)
Comment (Zacklane) [reply app/Resources/views/security/login.html.twig]: Sorry. I've left it in my view blueprint and've forgotten to remove before push. Fixed.

File: composer.lock (hunk 6 above)
Comment (Zacklane) [reply composer.lock]: composer.lock removed.

File: app/Resources/views/security/login.html.twig (hunk 4 above)
Comment (Zacklane) [reply app/Resources/views/security/login.html.twig]: Changed tag formatting in my each previous commits as you've required.

File: app/Resources/views/base.html.twig [hunk 10]
```diff
@@ -4,26 +4,37 @@
...
     <link rel="stylesheet" href="{{ asset('assets/css/mpcrm.css') }}" media="all">
-    <link rel="shortcut icon" href="/bundles/app/favicon.ico" type="image/x-icon" />
+    <link rel="shortcut icon" href="/bundles/app/favicon.ico" type="image/x-icon"/>
     <link rel="icon" href="/bundles/app/favicon.ico" type="image/x-icon">
 </head>
 <body class="aui-page-fixed">
 <div id="page">
     <header id="header" role="banner">
-        <nav class="aui-header" role="navigation">
+        <nav class="aui-header" role="navigation" style="min-height: 40px">
```
Comment (thecatontheflat) [inline app/Resources/views/base.html.twig]: Move inline style to the appropriate file:
https://github.com/thecatontheflat/atlassian-vendor-tools/blob/master/web/assets/css/mpcrm.css

File: app/Resources/views/base.html.twig [hunk 11]
```diff
@@ -34,9 +45,10 @@
                 </div>
             </div>
         </header>
-        <div class="aui-page-panel">
+        <div class="aui-page-panel" style="min-height: 400px">
```
Comment (thecatontheflat) [inline app/Resources/views/base.html.twig]: Same here, goes here:
https://github.com/thecatontheflat/atlassian-vendor-tools/blob/master/web/assets/css/mpcrm.css

File: app/Resources/views/security/login.html.twig [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+    <div class="avt-login-wrap">
+        {% if error %}
+            <div>{{ error.messageKey|trans(error.messageData, 'security') }}</div>
+        {% endif %}
+
+        <form action="{{ path("login") }}" method="post" class="aui">
+
+            <div class="field-group">
+                <label for="username">Username<span class="aui-icon icon-required">(required)</span></label>
+                <input type="text" id="username" name="_username" value="{{ last_username }}" required="required"
```
Comment (thecatontheflat) [inline app/Resources/views/security/login.html.twig]: Format it to be in one line

File: app/Resources/views/security/login.html.twig [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+            </div>
+
+            <div class="field-group">
+                <label for="password">Password<span class="aui-icon icon-required">(required)</span></label>
+                <input type="password" id="password" name="_password" required="required" class="text medium-field"/>
+            </div>
+            <fieldset class="group">
+                <div class="checkbox">
+                    <label for="remember_me"><input type="checkbox" id="remember_me" name="_remember_me"
+                                                    class="checkbox" value="on"/>Stay logged in</label>
```
Comment (thecatontheflat) [inline app/Resources/views/security/login.html.twig]: Same as above

**Review by thecatontheflat — CHANGES_REQUESTED** [review]

File: app/Resources/views/base.html.twig (hunk 11 above)
Comment (Zacklane) [reply app/Resources/views/base.html.twig]: Moved those inlines to mpcrm.css

File: app/Resources/views/security/login.html.twig (hunk 13 above)
Comment (Zacklane) [reply app/Resources/views/security/login.html.twig]: Sry. My IDE must be kidding... Fixed.

**Pull request merged**